# Likes Report

### Clare Rodgers
### September 10, 2025

## Section 1


- Who or what is the source of your personal Instagram data? (you downloaded the data, but did you create it?)

  **The source of the data is both myself and Meta. I was the person liking posts, the data here was gathered from my liked posts, but Meta was what kept track of the data for me to download it.** 
- Why was this data created and what purposes can the data serve?

  **This data was created to keep track of the posts that I like on Instagram. The purpose this data can serve is to help Instagram create an algorithm that best suits my interests. With an algorithm that aligns with my interests, I will be more likely to spend time on the app.**
- In what ways may this data be reliable?

  **This data may be reliable because I downloaded it right from Instagram. I hope that Meta would not tamper with the data.**
- In what ways may this data be unreliable?

   **This data may be unreliable because Meta could have potentially tampered with the data.**

## Section 2
Open and load json file

In [1]:
# These are imports we need
import json 
import pandas as pd
import matplotlib as plt

In [2]:
path = r"/Users/clarencerodgers/Downloads/instagram-silly_go0fy-2025-08-25-bAFcXbZj/your_instagram_activity/likes/liked_posts.json"
with open(path) as l:
    dict = json.load(l)

## Section 3
Convert the data into DataFrame object and assign it to a new variable using pandas DataFrame function

In [3]:
df = pd.DataFrame(dict)
df.head()

likes_media_likes
0  {'title': 'thekentstater', 'string_list_data':...
1  {'title': 'twilight.memes.forever', 'string_li...
2  {'title': 'dinoche.stinky', 'string_list_data'...
3  {'title': 'l3x1_lars0n', 'string_list_data': [...
4  {'title': 'microwavecorndog', 'string_list_dat...

In [4]:
dict.keys()

dict_keys(['likes_media_likes'])

In [5]:
df_tidy = pd.DataFrame(dict['likes_media_likes'])
df_tidy.head()

title                                   string_list_data
0           thekentstater  [{'href': 'https://www.instagram.com/p/DNx93w4...
1  twilight.memes.forever  [{'href': 'https://www.instagram.com/p/DNtWMcL...
2          dinoche.stinky  [{'href': 'https://www.instagram.com/reel/DK-c...
3             l3x1_lars0n  [{'href': 'https://www.instagram.com/reel/DLao...
4        microwavecorndog  [{'href': 'https://www.instagram.com/reel/DM9W...

## Section 4
Create a new column in DataFrame and assign timestamp values to each like in the new column

In [6]:
df_tidy['string_list_data'].iloc[0]

[{'href': 'https://www.instagram.com/p/DNx93w4wDVl/',
  'value': 'ð\x9f\x91\x8d',
  'timestamp': 1756135410}]

In [7]:
type(df_tidy['string_list_data'].iloc[0])

list

In [8]:
len(df_tidy['string_list_data'].iloc[0])

1

In [9]:
df_tidy['string_list_data'].iloc[0][0]

{'href': 'https://www.instagram.com/p/DNx93w4wDVl/',
 'value': 'ð\x9f\x91\x8d',
 'timestamp': 1756135410}

In [10]:
df_tidy['string_list_data'].iloc[0][0]['timestamp']

1756135410

In [11]:
df_tidy['Time of Like'] = [ i [0]['timestamp'] for i in df_tidy['string_list_data'] ]

df_tidy.head()

title                                   string_list_data  \
0           thekentstater  [{'href': 'https://www.instagram.com/p/DNx93w4...   
1  twilight.memes.forever  [{'href': 'https://www.instagram.com/p/DNtWMcL...   
2          dinoche.stinky  [{'href': 'https://www.instagram.com/reel/DK-c...   
3             l3x1_lars0n  [{'href': 'https://www.instagram.com/reel/DLao...   
4        microwavecorndog  [{'href': 'https://www.instagram.com/reel/DM9W...   

   Time of Like  
0    1756135410  
1    1756097990  
2    1756097922  
3    1756097892  
4    1756097830

In [12]:
# Good to have 
from datetime import datetime
df_tidy['Date of Like'] = [ datetime.fromtimestamp(i [0]['timestamp']) for i in df_tidy['string_list_data'] ]

In [13]:
df_tidy.head()

title                                   string_list_data  \
0           thekentstater  [{'href': 'https://www.instagram.com/p/DNx93w4...   
1  twilight.memes.forever  [{'href': 'https://www.instagram.com/p/DNtWMcL...   
2          dinoche.stinky  [{'href': 'https://www.instagram.com/reel/DK-c...   
3             l3x1_lars0n  [{'href': 'https://www.instagram.com/reel/DLao...   
4        microwavecorndog  [{'href': 'https://www.instagram.com/reel/DM9W...   

   Time of Like        Date of Like  
0    1756135410 2025-08-25 11:23:30  
1    1756097990 2025-08-25 00:59:50  
2    1756097922 2025-08-25 00:58:42  
3    1756097892 2025-08-25 00:58:12  
4    1756097830 2025-08-25 00:57:10

In [14]:
df_tidy['href'] = [ i [0]['href'] for i in df_tidy['string_list_data'] ]
df_tidy.head()

title                                   string_list_data  \
0           thekentstater  [{'href': 'https://www.instagram.com/p/DNx93w4...   
1  twilight.memes.forever  [{'href': 'https://www.instagram.com/p/DNtWMcL...   
2          dinoche.stinky  [{'href': 'https://www.instagram.com/reel/DK-c...   
3             l3x1_lars0n  [{'href': 'https://www.instagram.com/reel/DLao...   
4        microwavecorndog  [{'href': 'https://www.instagram.com/reel/DM9W...   

   Time of Like        Date of Like  \
0    1756135410 2025-08-25 11:23:30   
1    1756097990 2025-08-25 00:59:50   
2    1756097922 2025-08-25 00:58:42   
3    1756097892 2025-08-25 00:58:12   
4    1756097830 2025-08-25 00:57:10   

                                          href  
0     https://www.instagram.com/p/DNx93w4wDVl/  
1     https://www.instagram.com/p/DNtWMcLUDRd/  
2  https://www.instagram.com/reel/DK-cf6isitB/  
3  https://www.instagram.com/reel/DLaohRbvjIU/  
4  https://www.instagram.com/reel/DM9W5GHOGBh/

In [15]:
df_tidy['value'] = [ i [0]['value'] for i in df_tidy['string_list_data'] ]
df_tidy.head()

title                                   string_list_data  \
0           thekentstater  [{'href': 'https://www.instagram.com/p/DNx93w4...   
1  twilight.memes.forever  [{'href': 'https://www.instagram.com/p/DNtWMcL...   
2          dinoche.stinky  [{'href': 'https://www.instagram.com/reel/DK-c...   
3             l3x1_lars0n  [{'href': 'https://www.instagram.com/reel/DLao...   
4        microwavecorndog  [{'href': 'https://www.instagram.com/reel/DM9W...   

   Time of Like        Date of Like  \
0    1756135410 2025-08-25 11:23:30   
1    1756097990 2025-08-25 00:59:50   
2    1756097922 2025-08-25 00:58:42   
3    1756097892 2025-08-25 00:58:12   
4    1756097830 2025-08-25 00:57:10   

                                          href value  
0     https://www.instagram.com/p/DNx93w4wDVl/  👍  
1     https://www.instagram.com/p/DNtWMcLUDRd/  👍  
2  https://www.instagram.com/reel/DK-cf6isitB/  👍  
3  https://www.instagram.com/reel/DLaohRbvjIU/  👍  
4  https://www.instagram.com/reel/DM9W5GHOGBh/  👍

In [16]:
df_tidy.drop(columns = ['string_list_data'])

title  Time of Like        Date of Like  \
0               thekentstater    1756135410 2025-08-25 11:23:30   
1      twilight.memes.forever    1756097990 2025-08-25 00:59:50   
2              dinoche.stinky    1756097922 2025-08-25 00:58:42   
3                 l3x1_lars0n    1756097892 2025-08-25 00:58:12   
4            microwavecorndog    1756097830 2025-08-25 00:57:10   
...                       ...           ...                 ...   
19726            sukunacurse_    1724614781 2024-08-25 15:39:41   
19727              simsuicide    1724614737 2024-08-25 15:38:57   
19728              reachyusuf    1724614720 2024-08-25 15:38:40   
19729            okay.kay.dee    1724613738 2024-08-25 15:22:18   
19730               reachabed    1724613641 2024-08-25 15:20:41   

                                              href value  
0         https://www.instagram.com/p/DNx93w4wDVl/  👍  
1         https://www.instagram.com/p/DNtWMcLUDRd/  👍  
2      https://www.instagram.com/reel/DK-cf6isitB/  👍  
3      https://www.instagram.com/reel/DLaohRbvjIU/  👍  
4      https://www.instagram.com/reel/DM9W5GHOGBh/  👍  
...                                            ...   ...  
19726  https://www.instagram.com/reel/C76tJt9Pqh4/  👍  
19727  https://www.instagram.com/reel/C82aXgCOySy/  👍  
19728  https://www.instagram.com/reel/C97OPLvI9af/  👍  
19729     https://www.instagram.com/p/C_GZoUNu1Pc/  👍  
19730  https://www.instagram.com/reel/C_GkNQfN4SR/  👍  

[19731 rows x 5 columns]

## Section 5 

In [17]:
df_tidy.groupby('title').count().sort_values('value', ascending = False)

string_list_data  Time of Like  Date of Like  href  value
title                                                                         
kentwired                         195           195           195   195    195
dannyisinyourphone                183           183           183   183    183
destinydecktarot                  135           135           135   135    135
danhentschel                      128           128           128   128    128
reachyusuf                         88            88            88    88     88
...                               ...           ...           ...   ...    ...
hopelesscore01                      1             1             1     1      1
hopelesscore.fuelup                 1             1             1     1      1
hopelessandthriving                 1             1             1     1      1
hopeinherheart_                     1             1             1     1      1
zzzzeyk                             1             1             1     1      1

[10670 rows x 5 columns]

## Section 6

### Hypothesis:

Theoretical: I like more posts before I go to bed than during any other time of day.

Statistical: H0 (null): My liked posts are evenly distributed throughout the day.
H1: I like more posts during night hours (11pm-2am) than I do any other time of day.

My statistical hypothesis can be tested by looking at the data in the json files. My theoretical hypothesis is more abstract because I do not specify a timeframe.

This is my hypothesis because as observed in the first 5 rows of the DataFrame in Section 4 (Date of Like column), 4 out of 5 of my liked posts were liked during the 0th hour (12am). 

## Section 7

To help me test my hypothesis, I am going to use the saved posts json file. I think this json file contains posts that I have added to my saved posts on Instagram, as well as the time I saved them. Saved posts are used to help you find posts easier on Instagram. I usually save posts that I really like so I think this will help me to test my hypothesis.

## Section 8

Open and load json file

In [18]:
path = r"/Users/clarencerodgers/Downloads/instagram-silly_go0fy-2025-08-25-bAFcXbZj/your_instagram_activity/saved/saved_posts.json"
with open(path) as s:
    dict = json.load(s)

## Section 9

In [19]:
df = pd.DataFrame(dict)
df.head()

saved_saved_media
0  {'title': 'microwavecorndog', 'string_map_data...
1  {'title': 'bakedrizzledust', 'string_map_data'...
2  {'title': 'meme_hub1112', 'string_map_data': {...
3  {'title': 'typeoflarry', 'string_map_data': {'...
4  {'title': 'thenokidscollective', 'string_map_d...

In order to test my hypothesis, I am going to try to tidy up the table so I can see the date and time of the saved post.

In [20]:
dict.keys()

dict_keys(['saved_saved_media'])

In [21]:
df_tidy = pd.DataFrame(dict['saved_saved_media'])
df_tidy.head()

title                                    string_map_data
0     microwavecorndog  {'Saved on': {'href': 'https://www.instagram.c...
1      bakedrizzledust  {'Saved on': {'href': 'https://www.instagram.c...
2         meme_hub1112  {'Saved on': {'href': 'https://www.instagram.c...
3          typeoflarry  {'Saved on': {'href': 'https://www.instagram.c...
4  thenokidscollective  {'Saved on': {'href': 'https://www.instagram.c...

I am using .iloc[] to make it so that the timestamp will have its own column in the table

In [22]:
df_tidy['string_map_data'].iloc[0]

{'Saved on': {'href': 'https://www.instagram.com/reel/DM9W5GHOGBh/',
  'timestamp': 1756097843}}

In [23]:
type(df_tidy['string_map_data'].iloc[0])

dict

In [24]:
len(df_tidy['string_map_data'].iloc[0])

1

In [26]:
df_tidy['string_map_data'].iloc[0]['Saved on']['timestamp']

1756097843

In [32]:
df_tidy['Time of Save'] = [i['Saved on']['timestamp'] for i in df_tidy['string_map_data']]

df_tidy.head()

title                                    string_map_data  \
0     microwavecorndog  {'Saved on': {'href': 'https://www.instagram.c...   
1      bakedrizzledust  {'Saved on': {'href': 'https://www.instagram.c...   
2         meme_hub1112  {'Saved on': {'href': 'https://www.instagram.c...   
3          typeoflarry  {'Saved on': {'href': 'https://www.instagram.c...   
4  thenokidscollective  {'Saved on': {'href': 'https://www.instagram.c...   

   Time of Like  Time of Save  
0    1756097843    1756097843  
1    1756065747    1756065747  
2    1755985051    1755985051  
3    1755984816    1755984816  
4    1755984640    1755984640

(I accidentally named the column Time of Like the first time instead of Time of Save) 

I am now going to import datetime so that I can find the exact date and time of the save

In [33]:
from datetime import datetime
df_tidy['Date of Save'] = [datetime.fromtimestamp(i['Saved on']['timestamp']) for i in df_tidy['string_map_data']]

In [34]:
df_tidy.head()

title                                    string_map_data  \
0     microwavecorndog  {'Saved on': {'href': 'https://www.instagram.c...   
1      bakedrizzledust  {'Saved on': {'href': 'https://www.instagram.c...   
2         meme_hub1112  {'Saved on': {'href': 'https://www.instagram.c...   
3          typeoflarry  {'Saved on': {'href': 'https://www.instagram.c...   
4  thenokidscollective  {'Saved on': {'href': 'https://www.instagram.c...   

   Time of Like  Time of Save        Date of Save  
0    1756097843    1756097843 2025-08-25 00:57:23  
1    1756065747    1756065747 2025-08-24 16:02:27  
2    1755985051    1755985051 2025-08-23 17:37:31  
3    1755984816    1755984816 2025-08-23 17:33:36  
4    1755984640    1755984640 2025-08-23 17:30:40

Now that I can see the date and times of the saved posts, I am going to try to tidy up the table just a bit more to make it more organized

In [35]:
df_tidy['href'] = [i ['Saved on']['href'] for i in df_tidy['string_map_data']]
df_tidy.head()

title                                    string_map_data  \
0     microwavecorndog  {'Saved on': {'href': 'https://www.instagram.c...   
1      bakedrizzledust  {'Saved on': {'href': 'https://www.instagram.c...   
2         meme_hub1112  {'Saved on': {'href': 'https://www.instagram.c...   
3          typeoflarry  {'Saved on': {'href': 'https://www.instagram.c...   
4  thenokidscollective  {'Saved on': {'href': 'https://www.instagram.c...   

   Time of Like  Time of Save        Date of Save  \
0    1756097843    1756097843 2025-08-25 00:57:23   
1    1756065747    1756065747 2025-08-24 16:02:27   
2    1755985051    1755985051 2025-08-23 17:37:31   
3    1755984816    1755984816 2025-08-23 17:33:36   
4    1755984640    1755984640 2025-08-23 17:30:40   

                                          href  
0  https://www.instagram.com/reel/DM9W5GHOGBh/  
1  https://www.instagram.com/reel/DNWFK70vQJf/  
2  https://www.instagram.com/reel/DNbMrwXzjsz/  
3  https://www.instagram.com/reel/DMI9ABJMI-z/  
4  https://www.instagram.com/reel/DNgw-8qTlxT/

Now that the data is more organized, I am going to display the first 10 rows of the data. 

In [39]:
df_tidy.head(10)

title                                    string_map_data  \
0     microwavecorndog  {'Saved on': {'href': 'https://www.instagram.c...   
1      bakedrizzledust  {'Saved on': {'href': 'https://www.instagram.c...   
2         meme_hub1112  {'Saved on': {'href': 'https://www.instagram.c...   
3          typeoflarry  {'Saved on': {'href': 'https://www.instagram.c...   
4  thenokidscollective  {'Saved on': {'href': 'https://www.instagram.c...   
5      lolpokemonedits  {'Saved on': {'href': 'https://www.instagram.c...   
6        illeatpikachu  {'Saved on': {'href': 'https://www.instagram.c...   
7         nikarrr_hats  {'Saved on': {'href': 'https://www.instagram.c...   
8            disrvspec  {'Saved on': {'href': 'https://www.instagram.c...   
9              n2mem__  {'Saved on': {'href': 'https://www.instagram.c...   

   Time of Like  Time of Save        Date of Save  \
0    1756097843    1756097843 2025-08-25 00:57:23   
1    1756065747    1756065747 2025-08-24 16:02:27   
2    1755985051    1755985051 2025-08-23 17:37:31   
3    1755984816    1755984816 2025-08-23 17:33:36   
4    1755984640    1755984640 2025-08-23 17:30:40   
5    1755984133    1755984133 2025-08-23 17:22:13   
6    1755984048    1755984048 2025-08-23 17:20:48   
7    1755983324    1755983324 2025-08-23 17:08:44   
8    1755982997    1755982997 2025-08-23 17:03:17   
9    1755982741    1755982741 2025-08-23 16:59:01   

                                          href  
0  https://www.instagram.com/reel/DM9W5GHOGBh/  
1  https://www.instagram.com/reel/DNWFK70vQJf/  
2  https://www.instagram.com/reel/DNbMrwXzjsz/  
3  https://www.instagram.com/reel/DMI9ABJMI-z/  
4  https://www.instagram.com/reel/DNgw-8qTlxT/  
5  https://www.instagram.com/reel/DM4NarQzlSn/  
6  https://www.instagram.com/reel/DMekL6qy_Gj/  
7  https://www.instagram.com/reel/DNPAfzGTf9s/  
8  https://www.instagram.com/reel/DNtSeSt4rnT/  
9  https://www.instagram.com/reel/DNU3rS0IH2q/

To help summarize the data of saved posts, I am going to use groupby() and count()

In [43]:
df_tidy.groupby('title').count().sort_values('Time of Like', ascending = False)

string_map_data  Time of Like  Time of Save  \
title                                                                      
barbequebaconburgereveryday                6             6             6   
amitkumar_11ak                             5             5             5   
noahwaybabe                                5             5             5   
meas.says                                  5             5             5   
dannyisinyourphone                         5             5             5   
...                                      ...           ...           ...   
hadoublesan                                1             1             1   
hannahbrown.mp3                            1             1             1   
hattie_willoughby7                         1             1             1   
headvempty                                 1             1             1   
zonekx_444                                 1             1             1   

                             Date of Save  href  
title                                            
barbequebaconburgereveryday             6     6  
amitkumar_11ak                          5     5  
noahwaybabe                             5     5  
meas.says                               5     5  
dannyisinyourphone                      5     5  
...                                   ...   ...  
hadoublesan                             1     1  
hannahbrown.mp3                         1     1  
hattie_willoughby7                      1     1  
headvempty                              1     1  
zonekx_444                              1     1  

[980 rows x 5 columns]

## Section 10

I think the DataFrame represents different pieces of information about my saved posts. The title column seems to be the name of the account that created the post I saved. The Time of Save and Date of Save columns represent the date and time I saved the post (however I do not understand how to interpret the numbers in the Time of Save column). The href column is a link to the post I saved. I am not sure what the string_map_data is. I think it might be all the information that the data stored about my saved posts. 

I do not think my DataFrame of my saved posts meets tidy principles. This is because I could not get the values column. I tried my best to make it as tidy as possible, but I could not get the values column into my DataFrame like we did in Section 4. I am also not sure what the value column represents. 

I think the DataFrame created from my saved posts data can be used to test my hypothesis because I was able to get the time and date of each saved post. 

My hypothesis was I like more posts during night hours (11pm-2am) than I do any other time of day. The first 10 rows of data from my saved posts (In Section 9 of this report) do not support this hypothesis, as 9 out of 10 saved posts were saved during the 16th and 17th hours of the day (4pm and 5pm), which does not fall into the 11pm-2am time window. 

However, this does not necessarily mean that my hypothesis is entirely false, as this only shows 10 rows out of 980 rows of the data from my saved posts. Another approach to test this hypothesis could be to look at other rows within the data. Maybe the next steps to test this hypothesis could be to look at the last 10 rows of the data, or to find a way to look at some of the rows in the middle of the data. 
